src/Entities/Entity.cpp

```diff
@@ -222,8 +222,15 @@ void cEntity::Destroy(bool a_ShouldBroadcast)
 	}
 
 	cChunk * ParentChunk = GetParentChunk();
-	m_World->QueueTask([this, ParentChunk](cWorld & a_World)
+
+	// We remove the entity from memory 5 ticks later, giving everyone plenty of time
```

Comment:
I wanted to ask you about this.

Previously, we had code like `If (m_Target->IsDestroyed()) m_Target = nullptr`, but that was troublesome because sometimes m_Target is already freed from memory and basically calling this is UB and there were occasional crashes.

Back then, my solution was the complex "cPawn::StopEveryoneFromTargetingMe" and several other functions that nullify m_Target when needed. But looking back it is seemingly overkill, if we just postpone freeing the entity for 2 or more ticks (5 was just arbitrary), that everyone gets a chance to tick and check `IsDestroyed` (now replaced with IsTicking) and nullify the pointers and the problem is solved, is it not?

Unless of course there was another reason for my complex solution, but I do not remember any.